CMakeLists.txt

```diff
@@ -35,9 +35,23 @@ endif()
 
 project(msquic)
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_C_STANDARD 11)
```

Comment:
I'd prefer to not use these globals, they cause dependency issues. Instead, if we want to set standard this way (I'm fine with it) set it as a public target property on the `inc` target. That way its done the correct way.

Reply:
If I do something like this
```
set_property(TARGET inc PROPERTY CXX_STANDARD 17)
```
I get 
```
CMake Error at src/inc/CMakeLists.txt:15 (set_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_STANDARD" is not allowed.
```

Reply:
Ah, but this seems to work:
```
target_compile_features(inc INTERFACE cxx_std_17)
target_compile_features(inc INTERFACE c_std_11)
```